ElementX/Sources/Screens/RoomScreen/View/KnockRequestsBannerView.swift

```diff
@@ -0,0 +1,165 @@
+        guard let first = requests.first else {
+            return ""
+        }
+        
+        let string = first.displayName ?? first.userID
+        return L10n.tr("Localizable", "screen_room_multiple_knock_requests_title", string, avatars.count - 1)
+    }
+    
+    var body: some View {
+        mainContent
+            .padding(16)
+            .background(.compound.bgCanvasDefault, in: RoundedRectangle(cornerRadius: 12))
+            .compositingGroup()
+            .shadow(color: Color(red: 0.11, green: 0.11, blue: 0.13).opacity(0.1), radius: 12, x: 0, y: 4)
+            .padding(.horizontal, 16)
+    }
+    
+    @ViewBuilder
+    private var mainContent: some View {
+        if requests.count == 1,
+           let request = requests.first {
+            singleRequestView(request: request)
+        } else {
+            multipleRequestsView
+        }
+    }
+    
+    private func singleRequestView(request: KnockRequestInfo) -> some View {
+        VStack(spacing: 14) {
+            HStack(spacing: 10) {
+                LoadableAvatarImage(url: request.avatarURL,
+                                    name: request.displayName,
+                                    contentID: request.userID,
+                                    avatarSize: .user(on: .knockingUser), mediaProvider: mediaProvider)
+                VStack(spacing: 0) {
+                    HStack(alignment: .top, spacing: 0) {
+                        Text(L10n.screenRoomSingleKnockRequestTitle(request.displayName ?? request.userID))
+                            .lineLimit(2)
+                            .font(.compound.bodyMDSemibold)
+                            .foregroundStyle(.compound.textPrimary)
+                            .frame(maxWidth: .infinity, alignment: .leading)
+                        dismissButton
+                    }
+                    if request.displayName != nil {
+                        Text(request.userID)
+                            .lineLimit(2)
+                            .font(.compound.bodySM)
+                            .foregroundStyle(.compound.textSecondary)
+                            .frame(maxWidth: .infinity, alignment: .leading)
+                    }
+                }
+            }
+            if let reason = request.reason {
+                Text(reason)
+                    .lineLimit(2)
+                    .font(.compound.bodyMD)
+                    .foregroundStyle(.compound.textPrimary)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+            }
+            HStack(spacing: 12) {
+                Button(L10n.screenRoomSingleKnockRequestViewButtonTitle, action: {
+                    onViewAll()
+                })
+                .buttonStyle(.compound(.secondary))
+                Button(L10n.screenRoomSingleKnockRequestAcceptButtonTitle, action: {
+                    onViewAll()
+                })
+                .buttonStyle(.compound(.primary))
```

Comment:
You can simplify this to pass the action directly
```suggestion
                Button(L10n.screenRoomSingleKnockRequestViewButtonTitle, action: onViewAll)
                    .buttonStyle(.compound(.secondary))
                Button(L10n.screenRoomSingleKnockRequestAcceptButtonTitle, action: onViewAll)
                    .buttonStyle(.compound(.primary))
```

Reply:
Also, are both of these buttons meant to have the same action?